apis/addons/v1alpha1/addons_types.go

```diff
@@ -5,18 +5,40 @@ import (
 )
 
 // AddonSpec defines the desired state of Addon.
-type AddonSpec struct{}
+type AddonSpec struct {
+	Namespaces []AddonNamespace `json:"namespaces,omitempty"`
```

Comment:
Let's try to add at least a little documentation to every parameter:

Suggestion:
```go
// AddonSpec defines the desired state of an Addon.
type AddonSpec struct {
	// Defines a list of Kubernetes Namespaces that belong to this Addon.
	// Namespaces listed here will be created prior to installation of the Addon and
	// will be removed from the cluster when the Addon is deleted.
	// Collisions with existing Namespaces are NOT allowed.
	Namespaces []AddonNamespace `json:"namespaces,omitempty"`
}

// AddonNamespace defines a Namespace belonging to an Addon.
type AddonNamespace struct {
	// Name of the KubernetesNamespace.
	Name string `json:"name"`
}
```

Reply:
good catches! :+1: